include/lp_hash_table.h

```diff
@@ -84,6 +84,7 @@ template <typename T> struct LpHashTable {
    * @return. nothing. we are sure that we can insert
    */
   void insert(const char *key, T value) {
+    assert(key != NULL && value != NULL);
```

Comment:
Do it in separate asserts. I know this seems like nitpicking but you'll it very useful when one assert fires, to know immediately which thing was the problem.